sentry-android-fragment/src/main/java/io/sentry/android/fragment/FragmentLifecycleIntegration.kt

```diff
@@ -0,0 +1,66 @@
+package io.sentry.android.fragment
+
+import android.app.Activity
+import android.app.Application
+import android.app.Application.ActivityLifecycleCallbacks
+import android.os.Bundle
+import androidx.fragment.app.FragmentActivity
+import io.sentry.IHub
+import io.sentry.ILogger
+import io.sentry.Integration
+import io.sentry.SentryLevel.DEBUG
+import io.sentry.SentryOptions
+import java.io.Closeable
+
+class FragmentLifecycleIntegration(
+    private val application: Application,
+    private val logger: ILogger
+) : ActivityLifecycleCallbacks, Integration, Closeable {
+
+    private lateinit var hub: IHub
+
+    override fun register(hub: IHub, options: SentryOptions) {
+        this.hub = hub
+
+        application.registerActivityLifecycleCallbacks(this)
+        logger.log(DEBUG, "FragmentLifecycleIntegration installed.")
+    }
+
+    override fun close() {
+        application.unregisterActivityLifecycleCallbacks(this)
+        logger.log(DEBUG, "FragmentLifecycleIntegration removed.")
+    }
+
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        (activity as? FragmentActivity)
+            ?.supportFragmentManager
+            ?.registerFragmentLifecycleCallbacks(
+                SentryFragmentLifecycleCallbacks(hub),
+                true
+            )
+    }
+
+    override fun onActivityStarted(activity: Activity) {
+        // no-op
+    }
+
+    override fun onActivityResumed(activity: Activity) {
+        // no-op
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        // no-op
+    }
+
+    override fun onActivityStopped(activity: Activity) {
+        // no-op
+    }
+
+    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
+        // no-op
+    }
+
+    override fun onActivityDestroyed(activity: Activity) {
```

Comment:
I don't unregister `SentryFragmentLifecycleCallback` as stated in docs:

> All registered callbacks will be automatically unregistered when this FragmentManager is destroyed.

[Source](https://developer.android.com/reference/androidx/fragment/app/FragmentManager?hl=id#registerFragmentLifecycleCallbacks(androidx.fragment.app.FragmentManager.FragmentLifecycleCallbacks,%20boolean))

We probably want a callbacks to be registered for the whole lifecycle of the `FragmentManager`.

Reply:
good point, maybe we can add it as a comment and linking to the docs, just for reference

Reply:
Sure, good idea. Comment added in a1344ffe99dec542bf3ea10088191a00ae48c42f